at_client/lib/src/service/notification_service_impl.dart

```diff
@@ -180,43 +186,69 @@ class NotificationServiceImpl
   }
 
   @override
-  Future<NotificationResult> notify(
-      NotificationParams notificationParams,
-      { bool waitForFinalDeliveryStatus = true, // this was the behaviour before introducing this parameter
-        bool checkForFinalDeliveryStatus = true, // this was the behaviour before introducing this parameter
-        Function(NotificationResult)? onSuccess,
-        Function(NotificationResult)? onError,
-        Function(NotificationResult)? onSentToSecondary}) async {
+  Future<NotificationResult> notify(NotificationParams notificationParams,
+      {bool waitForFinalDeliveryStatus =
+          true, // this was the behaviour before introducing this parameter
+      bool checkForFinalDeliveryStatus =
+          true, // this was the behaviour before introducing this parameter
+      Function(NotificationResult)? onSuccess,
+      Function(NotificationResult)? onError,
+      Function(NotificationResult)? onSentToSecondary}) async {
     var notificationResult = NotificationResult()
       ..notificationID = notificationParams.id
       ..atKey = notificationParams.atKey;
     try {
-      // Notifies key to another notificationParams.atKey.sharedWith atsign
-      await _atClient.notifyChange(notificationParams);
+      // If sharedBy atSign is null, default to current atSign.
+      if (notificationParams.atKey.sharedBy.isNull) {
+        notificationParams.atKey.sharedBy = _atClient.getCurrentAtSign();
+      }
+      // Append '@' if not already set.
+      AtUtils.formatAtSign(notificationParams.atKey.sharedBy!);
+      // validate notification request
+      atClientValidation.validateNotificationRequest(
+          notificationParams, _atClient.getPreferences()!);
+      // Get the EncryptionInstance to encrypt the data.
+      var atKeyEncryption = AtKeyEncryptionManager.get(
+          notificationParams.atKey, _atClient.getCurrentAtSign()!);
+      // Get the NotifyVerbBuilder from NotificationParams
+      var builder = await NotificationRequestTransformer(
+              _atClient.getCurrentAtSign()!,
+              _atClient.getPreferences()!,
+              atKeyEncryption)
+          .transform(notificationParams);
+
+      if (builder.atKey!.startsWith(AT_PKAM_PRIVATE_KEY) ||
```

Comment:
do we need this check? do we notify these pkam keys?

Reply:
No Murali, Will remove the IF condition and push the changes.